src/find-menu.js

```diff
@@ -26,42 +33,171 @@ class FindMenu extends HTMLElement {
 
     this.previousButton.addEventListener('click', () => {
       const { value } = this;
       if (!value) return;
-      this.dispatchEvent(new CustomEvent('previous', { detail: { value, findNext: false } }));
+      this.findInWebview(value, { forward: false, findNext: true });
     });
+    
     this.nextButton.addEventListener('click', () => {
       const { value } = this;
       if (!value) return;
-      this.dispatchEvent(new CustomEvent('next', { detail: { value, findNext: true } }));
+      this.findInWebview(value, { forward: true, findNext: true });
     });
+    
     this.hideButton.addEventListener('click', () => this.hide());
   }
 
+  setupWebviewNavigationListener() {
+    const webview = this.getWebviewElement();
+    if (!webview) return;
+
+    // Listen for did-navigate events to detect content type
+    webview.addEventListener('did-navigate', () => {
+      this.detectContentType();
+    });
+    
+    webview.addEventListener('did-navigate-in-page', () => {
+      this.detectContentType();
+    });
+    
+    // Also check when loading finishes
+    webview.addEventListener('did-finish-load', () => {
+      this.detectContentType();
+    });
+  }
+
+  async detectContentType() {
+    const webview = this.getWebviewElement();
+    if (!webview) return;
+    
+    try {
+      // Check if current page is a PDF by examining the URL or content
+      const url = await webview.getURL();
+      this.isPdf = url.toLowerCase().endsWith('.pdf') || 
+                   url.toLowerCase().includes('application/pdf');
+      
+      // If we need more precise detection, we can use executeJavaScript
+      if (!this.isPdf) {
+        const contentType = await webview.executeJavaScript(`
+          document.contentType || 
+          (document.querySelector('embed[type="application/pdf"]') ? 'application/pdf' : '')
+        `);
+        this.isPdf = contentType === 'application/pdf';
+      }
+    } catch (error) {
+      console.error('Error detecting content type:', error);
+    }
+  }
+  // fix (orignal code)
+  setupFoundInPageListener() {
+    // Get the webview element
+    const webview = this.getWebviewElement();
+    if (!webview) return;
+
+    // Listen for found-in-page events
+    webview.addEventListener('found-in-page', (event) => {
+      const { requestId, matches, activeMatchOrdinal } = event.result;
+      
+      // Ensure this is a response to our current request
+      if (requestId !== this.currentRequestId) return;
+      
+      // Update match count display
+      if (matches > 0) {
+        // For PDFs, handle the duplicate counting issue
+        if (this.isPdf && this.matchesCount > 0 && matches > this.matchesCount) {
+          // In PDFs, sometimes the count doubles incorrectly => keep original count
+          this.currentMatchIndex = (this.currentMatchIndex % this.matchesCount) + 1;
+        } else {
+          this.matchesCount = matches;
+          this.currentMatchIndex = activeMatchOrdinal;
```

Comment:
There seems to be an issue with how `currentMatchIndex` is updated when navigating backward. If the user is on match 3/10 and clicks "Previous," it should correctly show 2/10, but it's currently jumping to 4/10 due to incorrect index handling. Consider tracking the last search direction and adjusting `currentMatchIndex` accordingly.